feat: add support for Qt6-based builds

WereCatf opened the pull request:
With Mumble moving to use Qt6 this container was no longer able to build recent versions successfully. This commit adds support for building Qt6-based version of mumble-server.

WereCatf committed:
feat: add support for Qt6-based builds

Co-Authored-By: Robert Adam <dev@robert.adam.de>

JobberRT commented:
this build seems to have a peoblem:

<W>2024-11-30 04:37:28.893 No functional TLS backend was found

<F>2024-11-30 04:37:28.893 SSL: this version of Murmur is built against Qt without SSL Support. Aborting.

Krzmbrzl commented on scripts/install_qt.sh:
```diff
@@ -0,0 +1,52 @@
+	exit 1
+else
+	# 1726713847 refers to the timestamp of the commit that switched Mumble to use Qt6
+	if [ ${timestamp} -ge 1726713847 ]
```
Using the timestamp won't work. There will be backport commits to the 1.5 branch that are more recent than the Qt 6 introduction but still the 1.5 branch requires Qt5.

Instead, we'll have to decide based on the used Mumble version. Use Qt 6 for "latest" (or unspecified) and for everything 1.6 and above.

Krzmbrzl commented on Dockerfile:
```diff
@@ -1,76 +1,85 @@
 
 COPY --from=build /mumble/repo/build/mumble-server /usr/bin/mumble-server
 COPY --from=build /mumble/repo/default_config.ini /etc/mumble/bare_config.ini
+COPY --from=build /usr/lib/${ARCH}/qt6/plugins/tls /usr/lib/${ARCH}/qt6/plugins/tls
```
Why is this suddenly necessary when it wasn't before?

Krzmbrzl committed:
Update to Ubuntu 24.04

The Qt 6 version shipped in 22.04 is buggy when it comes to OpenSSL
detection. This leads to Qt not having SSL support, which effectively
makes the server unusable.

The SSL issues have been fixed in Qt 6.3 and newer, but 22.04 ships 6.2.

See also https://forum.qt.io/topic/156840/ssl-not-supported-after-migration-from-qt-5-to-qt-6-2-4/2

Krzmbrzl committed:
Adapt capitalization of "as" keywords

Krzmbrzl committed:
Enable unity builds

Krzmbrzl committed:
Increase test timeout

Due to some of the builds happening on an emulator, executing tests can
take longer than 5 minutes (the default timeout) per test case.

Krzmbrzl commented:
@JobberRT the issue you were seeing is a bug in Qt 6.2 that ships with Ubuntu 22.04. Using 24.04 that comes with Qt 6.5 resolves the issue as the underlying bug has been fixed in Qt 6.3.

Krzmbrzl commented:
Thanks for working on this :+1:
